src/odemis/util/comp.py

```diff
@@ -141,10 +141,15 @@ def generate_zlevels(focuser: "Actuator", zrange: Tuple[float, float], zstep: fl
         raise KeyError(f"The focus actuator {focuser} does not have z axis")
 
     focuser_pos = focuser.position.value["z"]
+    focuser_rng = focuser.axes["z"].range
+    if (focuser_pos > focuser_rng[1]) or (focuser_pos < focuser_rng[0]):
```

Comment:
The more "pythonic" way to write it is:
```suggestion
    if not focuser_rng[0] <= focuser_pos <= focuser_rng[1]:
```